dags/multipod/maxtext_profiling.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A DAG to run MaxText profiling tests.
+"""
+import datetime
+from airflow import models
+from dags import composer_env, test_owner, gcs_bucket
+from dags.vm_resource import TpuVersion, Zone, DockerImage
+from dags.multipod.configs import gke_config
+from dags.multipod.configs.common import SetupMode
+
+# Run once a day at 6 am UTC (10 pm PST)
+SCHEDULED_TIME = "0 6 * * *" if composer_env.is_prod_env() else None
+
+with models.DAG(
+    dag_id="maxtext_profiling",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "stable", "nightly"],
+    start_date=datetime.datetime(2024, 3, 1),
+    catchup=False,
+    concurrency=2,
+) as dag:
+  base_output_directory = gcs_bucket.XLML_OUTPUT_DIR
+  dataset_path = gcs_bucket.MAXTEXT_DIR
+  docker_images = [
+      (SetupMode.STABLE, DockerImage.MAXTEXT_JAX_STABLE),
+      (SetupMode.NIGHTLY, DockerImage.MAXTEXT_JAX_NIGHTLY),
+  ]
+
+  profiling_cmds = (
+      "export RUN_NAME='profiling_tests_$(date +%Y-%m-%d-%H-%M-%S)'",
```

Comment:
Can you double check that this is being formatted properly in the command? With `'`, the `$(date ...)` should be treated as a string literal instead of being evaluated and substituted. If it looks fine, we might have an issue with how we're escaping the bash commands in the infra.

Also, this could collide for the nightly and stable tests if they launch at the same time. Maybe move this inside the for loop and append the test mode, e.g. `f"export RUN_NAME=profiling_tests_$(date ...)_{mode.value}"`

Reply:
Yes, I could see the run_name being propagated properly: http://shortn/_JbijteLw8L

Sounds good, I will append the test mode.

Reply:
Ah, thanks for confirming! This is an issue then. Could you try removing the `'` for now? I'll work on a fix in the infra, don't want to accidentally break your tests in the future.

Reply:
It works without `'`, so the command in my test will look like: `f"export RUN_NAME=profiling_tests_{mode.value}_$(date +%Y-%m-%d-%H-%M-%S)"`